tests/integration/test_logcollector/test_configuration/test_basic_configuration_log_format.py

```diff
@@ -0,0 +1,317 @@
+
+macos_duplicated_config_path = Path(CONFIGURATIONS_PATH, 'wazuh_duplicated_macos_configuration.yaml')
+macos_duplicated_cases_path = Path(TEST_CASES_PATH, 'cases_basic_configuration_log_format_macos_duplicated.yaml')
+
+macos_no_defined_config_path = Path(CONFIGURATIONS_PATH, 'wazuh_no_defined_location_macos_configuration.yaml')
+macos_no_defined_cases_path = Path(TEST_CASES_PATH, 'cases_basic_configuration_log_format_macos_no_defined.yaml')
+
+win_config_path = Path(CONFIGURATIONS_PATH, 'wazuh_basic_configuration_log_format_location.yaml')
+win_cases_path = Path(TEST_CASES_PATH, 'cases_basic_configuration_log_format_win.yaml')
+
+test_configuration, test_metadata, test_cases_ids = configuration.get_test_cases_data(default_cases_path)
+
+folder_path = tempfile.gettempdir()
+location = os.path.join(folder_path, 'test.txt')
+for test in test_metadata:
+    if test['location'] and test['log_format'] != 'djb-multilog':
+        test['location'] = re.escape(location)
+for test in test_configuration:
+    if test['LOCATION'] and test['LOG_FORMAT'] != 'djb-multilog':
+        test['LOCATION'] = location
+
+test_configuration = configuration.load_configuration_template(default_config_path, test_configuration, test_metadata)
+
+test_macos_dup_configuration, test_macos_dup_metadata, test_macos_dup_cases_ids = configuration.get_test_cases_data(macos_duplicated_cases_path)
+test_macos_dup_configuration = configuration.load_configuration_template(macos_duplicated_config_path, test_macos_dup_configuration, test_macos_dup_metadata)
+
+test_macos_nodef_configuration, test_macos_nodef_metadata, test_macos_nodef_cases_ids = configuration.get_test_cases_data(macos_no_defined_cases_path)
+test_macos_nodef_configuration = configuration.load_configuration_template(macos_no_defined_config_path, test_macos_nodef_configuration, test_macos_nodef_metadata)
+
+test_win_configuration, test_win_metadata, test_win_cases_ids = configuration.get_test_cases_data(win_cases_path)
+test_win_configuration = configuration.load_configuration_template(win_config_path, test_win_configuration, test_win_metadata)
+
+
+if sys.platform == MACOS:
+    test_configuration += test_macos_dup_configuration
+    test_configuration += test_macos_nodef_configuration
+    test_metadata += test_macos_dup_metadata
+    test_metadata += test_macos_nodef_metadata
+    test_cases_ids += test_macos_dup_cases_ids
+    test_cases_ids += test_macos_nodef_cases_ids
+if sys.platform == WINDOWS:
+    test_configuration += test_win_configuration
+    test_metadata += test_win_metadata
+    test_cases_ids += test_win_cases_ids
+
+
+local_internal_options = {logcollector_configuration.LOGCOLLECTOR_REMOTE_COMMANDS: '1', logcollector_configuration.LOGCOLLECTOR_DEBUG: '2'}
+
+log_format_not_print_analyzing_info = ['command', 'full_command', 'eventlog', 'eventchannel', 'macos']
+
+# Test daemons to restart.
+daemons_handler_configuration = {'all_daemons': True}
+
```

Comment:
Remember to always leave two empty spaces between functions. This is required by python linter tools. Check this in all tests where needed.

Reply:
Done